File: backend/app/api/v1/endpoints/health.py
from fastapi import APIRouter

from app import __version__
from app.schemas.health import ComponentStatus, HealthResponse

router = APIRouter(tags=["health"])


@router.get("/health", response_model=HealthResponse)
async def health() -> HealthResponse:
    return HealthResponse(
        status="ok",
        version=__version__,
        components=[
            ComponentStatus(name="api", status="ok"),
        ],
    )

File: frontend/src/api/health.ts
import { useQuery } from '@tanstack/react-query';

import { apiFetch } from './client';

// Health probe shape from ``GET /api/v1/health``. The backend already
// returns this exact JSON (see ``app/schemas/health.py``); we mirror
// the shape locally so the typecheck catches drift.
export interface HealthResponse {
  status: string;
  version: string;
  environment: string;
}

export function useHealth() {
  return useQuery<HealthResponse>({
    queryKey: ['health'],
    queryFn: () => apiFetch<HealthResponse>('/health'),
    // Health is cheap and changing — show fresh state every minute.
    staleTime: 60_000,
  });
}
